pymc/ode/utils.py

```diff
@@ -103,10 +103,25 @@ def augment_system(ode_func, n_states, n_theta):
 
     # Get symbolic representation of the ODEs by passing tensors for y, t and theta
     yhat = ode_func(t_y, t_t, t_p[n_states:])
-    # Stack the results of the ode_func into a single tensor variable
-    if not isinstance(yhat, (list, tuple)):
-        yhat = (yhat,)
-    t_yhat = at.stack(yhat, axis=0)
+    if isinstance(yhat, at.TensorVariable):
+        if yhat.ndim == 0:
+            # Convert yhat from scalar to array
+            t_yhat = at.stack((yhat,), axis=0)
+        elif yhat.ndim == 1:
+            t_yhat = yhat
+        else:
+            raise ValueError(
+                f"The odefunc returned a {yhat.ndim}-dimensional tensor, but 0 or 1 dimensions were expected."
+            )
+    elif isinstance(yhat, np.ndarray):
```

Comment:
This check is a bit arbitrary, no? Was the original issue related to this?

Otherwise why aren't we checking if the tuple/list are not composed of ndarrays as well?

Reply:
On a second though, yes.
Maybe simply doing `at.atleast_1d(yhat)` and then checking `ndim==1`?
Do we even need to anticipate the `ndarray` case?

Reply:
I think it is possible that `ode_func` returns an `ndarray` . This is for example what was tried here https://discourse.pymc.io/t/return-value-for-2n-dimensional-ode-system/4141/3 . It looks like in this case `np.concatenate` threw an error, but maybe there are other operations which will cast a `TensorVariable` to an `ndarray` without throwing an error.

Reply:
That's not possible

Reply:
I removed the check for `np.ndarray` and I made the code more concise by using `at.atleast_1d` as suggested. Please let me know if there is anything else that needs to be done before this PR is good to go.

Reply:
Thanks! We're working to fix the CI on main and once that's done we can rebase your branch so we can have a green CI before merging